vissl/hooks/state_update_hooks.py

```diff
@@ -95,6 +96,17 @@ def on_phase_start(self, task: "tasks.ClassyTask") -> None:
             if hasattr(task.dataloaders[phase_type].sampler, "set_epoch"):
                 # task.phase_idx is current running phase id
                 task.dataloaders[phase_type].sampler.set_epoch(task.phase_idx)
+
+        # call set_epoch and for AirstoreDataset since it handles shuffle
+        # behavior internally
+        if hasattr(task.dataloaders[phase_type], "dataset"):
```

Comment:
question: do we want to test the phase_type=train or test? aka, there's no strict requirement for the `phase_type=train` only?

Reply:
yeah I think check phase_type=train make sense. I already clicked the land button. Will create a new PR for it